File: flutter_userapp/lib/controller/fda_controller.dart
import 'package:flutter_userapp/domain/fda/fda.dart';
import 'package:flutter_userapp/domain/fda/fda_repository.dart';
import 'package:get/get.dart';

class FdaController extends GetxController {
  FdaRepository fdaRepository = FdaRepository();

  Future<Fda> findById(String id) {
    return fdaRepository.findById(id);
  }

  Future<List<Fda>> findAll() {
    return fdaRepository.findAll();
  }
}

File: fda/src/main/java/com/kiswire/fda/web/FdaController.java
package com.kiswire.fda.web;

import java.util.List;

import org.springframework.data.domain.PageRequest;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import com.kiswire.fda.domain.Fda;
import com.kiswire.fda.domain.FdaRepository;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
@RestController
public class FdaController {

	private final FdaRepository fdaRepository;
	
	@CrossOrigin
	@GetMapping("/fda")
	public List<Fda> findAll(){
		return fdaRepository.findAll(PageRequest.of(1, 20)).getContent();
	}
	
	@CrossOrigin
	@GetMapping("/fda/{id}")
	public Fda findById(@PathVariable String id) {
		return fdaRepository.findById(id).get();
	}
}